File: CharityWebFrontend/src/services/ChatbotService.js
import api from './api';

const ChatbotService = {
    ask(question, projectId = null, conversationId = null) {
        return api.post('/chatbot/ask', {
            question,
            projectId,
            conversationId,
            locale: 'vi-VN',
        });
    },

    getSuggestions(projectId = null) {
        const params = projectId ? { projectId } : {};
        return api.get('/chatbot/suggestions', { params });
    },
};

export default ChatbotService;

File: CharityWebBackend/src/main/java/group3/project/charityweb/chatbot/service/ChatbotService.java
package group3.project.charityweb.chatbot.service;

import group3.project.charityweb.chatbot.dto.request.ChatAskRequest;
import group3.project.charityweb.chatbot.dto.response.ChatAskResponse;
import group3.project.charityweb.chatbot.dto.response.ChatSuggestionResponse;

public interface ChatbotService {
    ChatAskResponse ask(ChatAskRequest request, String clientId, boolean authenticated);
    ChatSuggestionResponse getSuggestions(String projectId);
}

